tests/purs/passing/DerivingFunctor.purs

```diff
@@ -3,34 +3,124 @@ module Main where
 import Prelude
 import Data.Eq (class Eq1)
 import Effect.Console (log)
+import Data.List (List(..), (:))
+import Data.Tuple (Tuple(..))
 import Test.Assert
 
-type MyRecord a = { myField :: a }
+type RecordFields f a =
+  { a :: a
+  -- ensure original field order (not order based on label)
+  -- is order used in fold
```

Comment:
This comment's a little out of place here, since order should not be observable by any `map` operation.